internal/cloudapi/server.go

```diff
@@ -182,7 +182,22 @@ func (server *Server) Compose(w http.ResponseWriter, r *http.Request) {
 		for name, packages := range packageSets {
 			pkgs, _, err := server.rpmMetadata.Depsolve(packages, repositories, distribution.ModulePlatformID(), arch.Name())
 			if err != nil {
-				http.Error(w, fmt.Sprintf("Failed to depsolve base packages for %s/%s/%s: %s", ir.ImageType, ir.Architecture, request.Distribution, err), http.StatusInternalServerError)
+				var error_type int
+				switch err.(type) {
```

Comment:
I think here we should look into the `DNFError.Kind` field and check what kind of error we got. A `DownloadError` for instance is definitely a 500 imo, but a `DespolveError` should be a 400.

Reply:
With this solution I am aiming at a proposition which does not involve string parsing. But I have no reservation to go this direction.

The only thing is that the type of the error is not necessarily `*rpmmd.DNFError` at this point. The return type of `server.rpmMetadata.Depsolve(..)` is `error` so any kind of error is possible.
It means that to have a string parsing we will have to keep the distinction between the type `*rpmmd.DNFError` that contains a `kind` variable and the rest of the possible errors. I believe that any of those possible errors can be 500 anyway.

Reply:
This makes sense I think, and if we'd ever want to expand it in future we'd just create types in go for those specific errors (like `rpmmd.DNFDepsolveError`) etc...